helm/druid/templates/broker/deployment.yaml

```diff
@@ -41,11 +41,15 @@ spec:
         app: {{ include "druid.name" . }}
         release: {{ .Release.Name }}
         component: {{ .Values.broker.name }}
-      {{- with .Values.broker.podAnnotations }}
       annotations:
-{{ toYaml . | indent 8 }}
+        druid.k8s.enablePatching: "true"
+      {{- with .Values.broker.podAnnotations }}
+        {{- toYaml . | nindent 8 }}
       {{- end }}
     spec:
+      {{- if .Values.broker.serviceAccount.create }}
```

Comment:
This blocks external creation of the serviceAccount. Most other charts allow you to bring your own serviceAccount which may also make reusing the same one for all druid service types easier.

I am wondering if we should default serviceAccount.name to unset and if create is true it uses what the default would be. Then if serviceAccount.name is set and serviceAccount.create is false we can still set the value here.

The charts I am used to using have a template like:

```
{{/*
Create the name of the forwarder service account to use
*/}}
{{- define "fluentd.forwarder.serviceAccountName" -}}
{{- if .Values.forwarder.serviceAccount.create -}}
    {{ default (printf "%s-forwarder" (include "common.names.fullname" .)) .Values.forwarder.serviceAccount.name }}
{{- else -}}
    {{ default "default" .Values.forwarder.serviceAccount.name }}
{{- end -}}
{{- end -}}
```

and then always set the serviceAccountName with the output of the template.

Reply:
I see what you're saying.  I have a bit of an opposite experience from you in terms of what other public widely used charts offer which is why I went this way but I'm not married to it.  I can create a helper definition which does the above sort of style.

Reply:
Fixed